packages/features/schedules/components/DateOverrideInputDialog.tsx

```diff
@@ -48,10 +49,38 @@ const DateOverrideForm = ({
       value[0].end.getUTCMinutes() === 0
   );
 
-  const [date, setDate] = useState<Dayjs | null>(value ? dayjs.utc(value[0].start) : null);
+  const [selectedDates, setSelectedDates] = useState<Dayjs[]>(value ? [dayjs.utc(value[0].start)] : []);
+
+  const onDateChange = (newDate: Dayjs) => {
+    if (!value && selectedDates.length === 1 && yyyymmdd(selectedDates[0]) !== yyyymmdd(newDate)) {
+      const currentSelectedDate = selectedDates[0];
+
+      const updatedSelectedDates: Dayjs[] = [];
+
+      // Determine the start and end dates for the loop
+      const startDate = currentSelectedDate.isBefore(newDate) ? currentSelectedDate : newDate;
+      const endDate = currentSelectedDate.isBefore(newDate) ? newDate : currentSelectedDate;
+
+      // Loop from startDate to endDate and add each date to updatedSelectedDates
+      let currentDate = startDate;
+
+      while (!currentDate.isAfter(endDate, "day")) {
+        if (!excludedDates || !excludedDates.includes(yyyymmdd(currentDate))) {
+          updatedSelectedDates.push(currentDate);
+        }
+        currentDate = currentDate.add(1, "day");
+      }
+
+      setSelectedDates(updatedSelectedDates);
+
+      return;
+    }
+
+    setSelectedDates([newDate]);
+  };
```

Comment:
In this function we're checking whether a user has selected a single date already, post that we verify whether it is an editing override on the basis of this we either select the range of dates between the selected dates or just select this single date.